dnstable/entry.c

```diff
@@ -203,28 +198,26 @@ fmt_rrtypes_union(ubuf *u, const struct dnstable_entry *e)
 }
 
 static void
-fmt_rrtypes_union_json(yajl_gen g, const struct dnstable_entry *e)
+fmt_rrtypes_union_json(ubuf *u, const struct dnstable_entry *e)
 {
-	yajl_gen_status status;
-
-	add_yajl_string(g, "rrtypes");
-
-	status = yajl_gen_array_open(g);
-	assert(status == yajl_gen_status_ok);
+	declare_json_value(u, "rrtypes", false);
+	JSON_ARR_START(u);
 
 	if (e->rrtype_map != NULL) {
 		rrtype_unpacked_set rrtype_set;
 		int n_rrtypes = rrtype_union_unpack(ubuf_data(e->rrtype_map), ubuf_size(e->rrtype_map), &rrtype_set);
-		if (n_rrtypes == -1)
-			add_yajl_string(g, "<failure>");
-		else {
-			for (int n = 0; n < n_rrtypes; n++)
-				assert(yajl_gen_status_ok == fmt_rrtype_json(g, rrtype_set.rrtypes[n]));
+		if (n_rrtypes == -1) {
+			const char *_fail = "<failure>";
+			JSON_STR_APPEND(u, _fail);
+		} else {
+			for (int n = 0; n < n_rrtypes; n++) {
+				fmt_rrtype_json(u, rrtype_set.rrtypes[n]);
+				JSON_ARR_DELIMIT(u, n, n_rrtypes);
```

Comment:
Macro defn here end up with an extra superfluous semi-colon at the end of the line (after the closing brace).